File: module_database/src/main/java/com/v/module_database/realm/dao/UserDao.java
package com.v.module_database.realm.dao;

import com.v.module_database.realm.model.User;

import java.util.List;

/**
 * ClassName: UserDao
 * Description:
 *
 * @author zhuming
 * @package_name com.v.module_database.realm
 * @date 2023/9/17 12:00
 */
public interface UserDao {
    //插入
    void insert(User user) throws Exception;

    //查询
    List<User> getAllUser() throws Exception;

    //更新
    User updateUser(User User) throws Exception;

    //根据条件修改
    void updateUserByUserId(String userId, String userName, int age) throws Exception;

    //删除
    void deleteUser(String userId) throws Exception;

    //异步插入
    void insertUserAsync(User user) throws Exception;

    void closeDB();
}

File: module_jetpack/src/main/java/com/v/module_jetpack/room/dao/UserDao.kt
package com.v.module_jetpack.room.dao

import androidx.room.*
import com.v.module_jetpack.room.table.User

/**
 * ClassName: UserDao
 * Description:
 *
 * @author zhuming
 * @package_name com.v.module_jetpack.room.database
 * @date 2023/7/31 22:24
 */
@Dao
interface UserDao {
    /**
     * 增加
     */
    @Insert
    fun insertUser(vararg user: User)

    /**
     * 删除指定用户
     */
    @Delete
    fun deleteUser(vararg user: User)

    /**
     * 删除所有用户
     */
    @Query("DELETE FROM user")
    fun deleteAllUser()

    /**
     * 修改
     */
    @Update
    fun updateUser(vararg user: User)

    /**
     * 修改
     */
    @Query("UPDATE user SET user_name= :userName WHERE user_id = :userId")
    fun updateUserByUserId(userId: Int, userName: String)

    /**
     * 根据用户id获取count
     */
    @Query("SELECT COUNT(*) FROM user WHERE user_id = :userId")
    fun getCountByUserId(userId: Int):Int

    /**
     * 查询所有
     */
    @Query("SELECT * FROM user")
    fun getAllUser(): List<User>

    /**
     * 根据用户名查询
     */
    @Query("SELECT * FROM user WHERE user_name IN (:userName)")
    fun getFindName(userName: String): User
}